zerver/views/webhooks/stripe.py

```diff
@@ -0,0 +1,129 @@
+        elif event_type == "order.payment_failed":
+            link = "https://dashboard.stripe.com/orders/"+payload["data"]["object"]["id"]
+            amount_string = amount(payload["data"]["object"]["amount"], payload["data"]["object"]["currency"])
+            body_template = "An order payment on order with id **[{object[id]}](" + link + ")** for **" + amount_string + "** has failed."
+            body = body_template.format(**(payload["data"]))
+        elif event_type == "order.payment_succeeded":
+            link = "https://dashboard.stripe.com/orders/"+payload["data"]["object"]["id"]
+            amount_string = amount(payload["data"]["object"]["amount"], payload["data"]["object"]["currency"])
+            body_template = "An order payment on order with id **[{object[id]}](" + link + ")** for **"\
+                            + amount_string + "** has succeeded."
+            body = body_template.format(**(payload["data"]))
+        elif event_type == "order.updated":
+            link = "https://dashboard.stripe.com/orders/"+payload["data"]["object"]["id"]
+            amount_string = amount(payload["data"]["object"]["amount"], payload["data"]["object"]["currency"])
+            body_template = "The order with id **[{object[id]}](" + link + ")** for **" + amount_string + "** has been updated."
+            body = body_template.format(**(payload["data"]))
+        elif event_type == "transfer.failed":
+            link = "https://dashboard.stripe.com/transfers/"+payload["data"]["object"]["id"]
+            amount_string = amount(payload["data"]["object"]["amount"], payload["data"]["object"]["currency"])
+            body_template = "The transfer with description **{object[description]}** and id **[{object[id]}]("\
+                            + link + ")** for amount **"\
+                            + amount_string + "** has failed."
+            body = body_template.format(**(payload["data"]))
+        elif event_type == "transfer.paid":
+            link = "https://dashboard.stripe.com/transfers/"+payload["data"]["object"]["id"]
+            amount_string = amount(payload["data"]["object"]["amount"], payload["data"]["object"]["currency"])
+            body_template = "The transfer with description **{object[description]}** and id **[{object[id]}]("\
+                            + link + ")** for amount **"\
+                            + amount_string + "** has been paid."
+            body = body_template.format(**(payload["data"]))
+    except KeyError as e:
+        body = "Missing key {} in JSON".format(str(e))
+
+    # send the message
+    check_send_message(user_profile, client, 'stream', [stream], topic, body)
+
+    return json_success()
+
+def amount(amount, currency):
+    # type: (int, str) -> str
+    # zero-decimal currencies
+    zeroDecimalCurrencies = ["bif", "djf", "jpy", "krw", "pyg", "vnd", "xaf", "xpf", "clp", "gnf", "kmf", "mga", "rwf", "vuv", "xof"]
+    if currency in zeroDecimalCurrencies:
```

Comment:
In Zulip, we spell things e.g. `zero_decimal_currencies`, not zeroDecimalCurrencies.